src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -240,18 +246,42 @@ public static Day parseDay(String dayAsStr) throws ParseException {
     /**
      * Parses a local time and checks whether it is valid.
      */
-    public static LocalTime parseLocalTime(String localTimeAsStr) throws ParseException {
+    public static LocalTime parseLocalTime(String localTimeAsStr, boolean isStartHour) throws ParseException {
         try {
             int hour = Integer.parseInt(localTimeAsStr);
-            if (!Arrays.asList(Timetable.START_TIMINGS).contains(hour)) {
-                throw new ParseException("Invalid time");
+            if (!TimeUtil.isValidHour(hour)) {
+                throw new ParseException(MESSAGE_INVALID_HOUR);
             }
-            return new LocalTime(hour, 0);
+            return isStartHour
+                ? parseStartHour(hour)
+                : parseEndHour(hour);
         } catch (NumberFormatException nfe) {
             throw new ParseException("Invalid time");
         }
     }
 
+    /**
+     * Parses a start time.
+     */
+    public static LocalTime parseStartHour(int hour) throws ParseException {
+        if (TimeUtil.isValidStartHour(hour)) {
```

Comment:
This part could use a little bit of happy pathing. The else block could be removed altogether. Or, rather, the throw should be the first thing you do in the if statement, then the actual return statement can be unindented.